src/Microsoft.VisualStudio.Threading.Tests/AsyncReaderWriterLockTests.cs

```diff
@@ -2509,6 +2509,83 @@ public void PrecancelledWriteLockAsyncRequestOnSTA()
             Assert.False(this.asyncLock.IsWriteLockHeld);
         }
 
+#if DESKTOP || NETCOREAPP2_0
```

Comment:
It seems like leaking and deadlocks could be detected on other platforms as well. Could we keep the test everywhere but use `#if` just around the MTA assertions?

Reply:
That sounds good...